testsuite/features/step_definitions/salt_steps.rb

```diff
@@ -271,15 +271,19 @@
   # Complicated code because the checkbox is not a <input type=checkbox> but an <i>
   xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-square-o']" if action == 'check'
   xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-check-square-o']" if action == 'uncheck'
-  puts "Selected formulas [beginning]:\n#{find('#chooseFormulas')['innerHTML']}"
+  # WORKAROUND
+  # DOM refreshes content of chooseFormulas element by accessing it. Then conditions are evaluated properly.
+  find('#chooseFormulas')['innerHTML']
   if all(:xpath, xpath_query, wait: DEFAULT_TIMEOUT).any?
-    puts "Selected formulas [if]:\n#{find('#chooseFormulas')['innerHTML']}"
-    raise "xpath: #{xpath_query} not found" unless find(:xpath, xpath_query, wait: DEFAULT_TIMEOUT).click
+    element = find(:xpath, xpath_query, match: :first ,wait: DEFAULT_TIMEOUT) if find('#chooseFormulas')
+    raise "xpath: #{xpath_query} not found" unless element.nil?    
```

Comment:
Interesting, I thought that rubocop complains if we don't add a newline after a raise.

Reply:
Remove the `if find(#chooseFormulas) i.e. revert to original code